src/api/base.ts

```diff
@@ -0,0 +1,59 @@
+import ky from 'ky';
+
+import { ERROR_MESSAGE } from '../utils/constants';
+import { ApiError } from '../types';
+
+// const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+const apiUrl = 'https://sportujspolu-api.onrender.com/api/v1/';
+
+export const api = ky.create({
+  prefixUrl: apiUrl,
+  headers: {},
+});
+
+export const privateApi = (token: string) =>
+  ky.create({
+    prefixUrl: apiUrl,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+export const apiPost = async <T>(
+  endpoint: string,
+  body: unknown,
+  token?: string,
+  onError?: (error: ApiError) => void,
```

Comment:
When you have more than 2-3 params, use object. You might accidentally swap two params with similar types